client/src/app/menu/menu.component.ts

```diff
@@ -68,6 +75,28 @@ export class MenuComponent implements OnInit {
     })
   }
 
+  get language () {
+    return this.languageChooserModal.getCurrentLanguage()
+  }
+
+  get videoLanguages (): string[] {
+    if (!this.user.videoLanguages) return [this.i18n('any language')]
+    return this.user.videoLanguages
+      .map(l => Object.values(pickBy(I18N_LOCALES, (_, key) => startsWith(key, l)))[0])
```

Comment:
Use `this.serverService.getVideoLanguages()` to get language translations instead.

Also, a complex code should not be a getter.